cpu/cc2538/periph/uart.c

```diff
@@ -276,28 +160,21 @@ static int init_base(uart_t uart, uint32_t baudrate)
     /* UART Enable */
     u->cc2538_uart_ctl.CTLbits.UARTEN = 1;
 
+    /* register callbacks */
+    uart_ctx[uart].rx_cb = rx_cb;
+    uart_ctx[uart].arg = arg;
```

Comment:
This is very dangerous: setting the callbacks after enabling the UART device can lead unwanted behavior: if the driver is initialized a second time (e.g. to change the baudrate), the global NVIC interrupt is already enabled. So we could get a call to a wrong isr context.

Solution a) just move the two lines above before the UARTEN
Solution b) disable the gloabl interrupt source at the beginning of the initialization

Reply:
I went for a). Less instructions and therefore code size...